ipa-core/src/net/test.rs

```diff
@@ -488,23 +510,72 @@ impl TestServer {
     pub fn builder() -> TestServerBuilder {
         TestServerBuilder::default()
     }
+
+    #[cfg(all(test, unit_test))]
+    pub async fn oneshot_success(
+        req: hyper::Request<axum::body::Body>,
+        handler: Arc<dyn RequestHandler<HelperIdentity>>,
+    ) -> bytes::Bytes {
+        let test_server = TestServerBuilder::<Helper>::default()
+            .with_request_handler(handler)
+            .build()
+            .await;
+        let resp = test_server.server.handle_req(req).await;
+        let status = resp.status();
+        assert_eq!(StatusCode::OK, status);
+
+        resp.into_body().collect().await.unwrap().to_bytes()
+    }
 }
 
-#[derive(Default)]
-pub struct TestServerBuilder {
-    handler: Option<Arc<dyn RequestHandler<HelperIdentity>>>,
+impl TestServer<Shard> {
+    #[cfg(all(test, unit_test))]
+    pub async fn oneshot(
+        req: hyper::Request<axum::body::Body>,
+        handler: Arc<dyn RequestHandler<ShardIndex>>,
+    ) -> hyper::Response<axum::body::Body> {
+        let test_server = TestServerBuilder::<Shard>::default()
+            .with_request_handler(handler)
+            .build()
+            .await;
+        test_server.server.handle_req(req).await
+    }
+
+    #[cfg(all(test, unit_test))]
+    pub async fn oneshot_success(
```

Comment:
Nit: I think it would be better to have this function just handle the parsing of the ok response so that it's more composable. An extension fn from Response.

E.g. this goes from `oneshot_success` to `parse_success` and is used like this:

```
TestServer::<Shard>::oneshot(req, handler).parse_success().await;
```

Reply:
probably, I wanted to mimic the existing `assert_success_with` behaviour. Will consider this change if I need to use it somewhere else